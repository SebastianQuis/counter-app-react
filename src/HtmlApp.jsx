import React from 'react'

import PropTypes from 'prop-types'

const nombre = 'Sebastian';
const user = {
    nombre: 'Sebasxd',
    apodo: 'El pepe',
    direccion: {
        calle: 'Junin',
        num: 150
    }
}

// CODIGO QUE FUNCIONA SOLO UNA VEZ 
const getNombre = () => `${user.nombre}`;
const numArr = [1,2,3,4,5,6,7]

export const HtmlApp = ({nombre, apodo, edad}) => {
    // CODIGO QUE FUNCIONA CADA VEZ QUE ACTUALICE
    // STATEFULL
    // console.log(props); // viene del componente padre
    return (
        <>
            <h1>Impresion de variables en HTML</h1>
            <h2> {nombre} </h2>
            <p> {apodo} </p>
            <p> {edad + 5} </p>
            {/* <code>{ JSON.stringify( user ) }</code>
            <p>
                { numArr.map( num => {
                    return <p> {num}: Seb </p>
                }) }
            </p>
            <p>{getNombre()}</p> */}
        </>
    )
}

// TODO - TIPOS DE DATOS PARA PARAMETROS DEL COMPONENETE
HtmlApp.propTypes = {
    nombre: PropTypes.string.isRequired,
    edad: PropTypes.number.isRequired,
}

// TODO - definir los valores de los proptypes.
HtmlApp.defaultProps = {
    apodo: 'Sin apodo',
    edad: 1,
    nombre: 'Sin nombre',
}



