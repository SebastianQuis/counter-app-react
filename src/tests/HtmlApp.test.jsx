import { render } from "@testing-library/react"
import { HtmlApp } from "../HtmlApp";


describe('Pruebas en <HTMLAPP/>', () => { 
     const nombre = "Name";
     const apodo = "Apodo";
     const edad = 2;


     // PARA COMPONENTES ESTÁTICOS  
     test('debe de hacer match con el snapshot', () => { 
          const {container} = render( <HtmlApp nombre={nombre} apodo={ apodo } edad={edad}/> );
          console.log(`CONTAINER:  ${container}`);
          expect( container ).toMatchSnapshot(); // la data 
     });

     test('debe de mostrar el titulo en h2', () => { 
          const { container, getByText, getByTestId } = render( <HtmlApp nombre={nombre} apodo={ apodo } edad={edad}/> );
          // expect( getByText( nombre ) ).toBeTruthy();

          // const h2 = container.querySelector('h2');
          // console.log(`H2 HTML: ${h2.innerHTML}`);
          // expect(h2.innerHTML).toContain(nombre);

          expect( getByTestId('test-title').innerHTML ).toContain(nombre); 
          // expect( getByTestId('test-title') ).toBeTruthy(); 
     });


     test('debe de mostrar el apodo', () => { 
          const { getAllByText } = render( 
               <HtmlApp 
                    apodo={apodo} 
                    nombre={nombre} 
                    edad={edad} 
               /> 
          )
     
          expect( getAllByText(apodo).length ).toBe(2);
      });

})


