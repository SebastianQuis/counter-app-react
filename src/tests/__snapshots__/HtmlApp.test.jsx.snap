// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HTMLAPP/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Impresion de variables en HTML
  </h1>
  <div
    data-testid="test-title"
  >
     
    Name
     
  </div>
  <p>
     
    Apodo
     
  </p>
  <p>
     
    Apodo
     
  </p>
  <p>
     
    7
     
  </p>
</div>
`;
